src/gleam/string.gleam

```diff
@@ -126,6 +127,64 @@ pub fn replace(
   |> string_builder.to_string
 }
 
+// Creates a new `String` by replacing the first occurrence of a given Regex.
+///
+/// ## Examples
+///
+/// ```gleam
+/// > use re <- try(regex.from_string("[.]+"))
+/// > replace_with_regex("www.example.com", each: re, with: "-")
+/// "www-example.com"
+/// ```
+///
+pub fn replace_with_regex(
+  in string: String,
+  each pattern: Regex,
+  with substitute: String,
+) -> String {
+  string
+  |> string_builder.from_string
+  |> regex_replace(each: pattern, with: string_builder.from_string(substitute))
```

Comment:
Do we need to do the string builder wrapping for the string and the substitute? Looks like from the Erlang docs it isn't needed.

Reply:
it's either that or using charlist, right?

Reply:
Ok, fixed